libsignal-service/src/provisioning/manager.rs

```diff
@@ -255,27 +245,35 @@ impl<P: PushService> ProvisioningManager<P> {
             format!("/v1/accounts/{}/code/{}", msg_type, phone_number)
         }
     }
+
+    fn auth_override(&self) -> HttpAuthOverride {
+        let credentials = ServiceCredentials {
+            uuid: None,
+            phonenumber: self.phone_number.clone(),
+            password: Some(self.password.clone()),
+            signaling_key: None,
+            device_id: None,
+        };
+        if let Some(auth) = credentials.authorization() {
+            HttpAuthOverride::Identified(auth)
+        } else {
+            HttpAuthOverride::NoOverride
+        }
+    }
```

Comment:
This might be a little bit surprising that after using ProvisitionManager the auth has changed. This means the client of provisioning would need to save the http auth to be able to restore it.

Reply:
I get what you're saying. On the other hand, our `PushService` is owned by the `ProvisionManager` and never returned, *and* all the `PushService` implementations clone their `Credentials`, and thus don't have shared mutability for them. This shouldn't be observable for the outside world as far as I can tell.

This does put a constraint on `PushService` implementors though, which we probably want to document. Given this new information, would you change your mind or do I merge as-is? Either is good to me.

Reply:
> Hmm, the PushService is not owned by ProvisionManager anymore since we give a &mut instance to it.

Makes a lot of sense to set overrides then. 👍️